### PR TITLE
Add monitor sample

### PR DESCRIPTION
**Pull request by chiranjeevi2776**
This sample sets up Wi-Fi in monitor mode, parses incoming packets, and displays statistics at regular intervals.
SHEL- 1941

### CODE REVIEW

**Comment by chiranjeevi2776** [comment]
@richabp, can you please review the README.rst. It needs to be incorporated into the 2.6.0 release.

**Comment by chiranjeevi2776** [comment]
@richabp, can you please correct me what could be the wrong in below error log.
/home/runner/work/sdk-nrf/sdk-nrf/ncs/nrf/doc/_build/nrf/src/releases_and_maturity/releases/release-notes-changelog.rst:667: WARNING: term not in glossary: 'Monitor'

I have changed doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst with below

> * Added the :ref:`wifi_monitor_sample` sample that demonstrates how to start a nRF70 Series device in :term:`Monitor` mode.

**Comment by richabp** [comment]
> @richabp, can you please correct me what could be the wrong in below error log. /home/runner/work/sdk-nrf/sdk-nrf/ncs/nrf/doc/_build/nrf/src/releases_and_maturity/releases/release-notes-changelog.rst:667: WARNING: term not in glossary: 'Monitor'
> 
> I have changed doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst with below
> 
> > * Added the :ref:`wifi_monitor_sample` sample that demonstrates how to start a nRF70 Series device in :term:`Monitor` mode.

To use the :term: directive, the term "Monitor" should be included in the Glossary here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/glossary.html.

**Comment by chiranjeevi2776** [comment]
> > @richabp, can you please correct me what could be the wrong in below error log. /home/runner/work/sdk-nrf/sdk-nrf/ncs/nrf/doc/_build/nrf/src/releases_and_maturity/releases/release-notes-changelog.rst:667: WARNING: term not in glossary: 'Monitor'
> > I have changed doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst with below
> > > * Added the :ref:`wifi_monitor_sample` sample that demonstrates how to start a nRF70 Series device in :term:`Monitor` mode.
> 
> To use the :term: directive, the term "Monitor" should be included in the Glossary here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/glossary.html.

Updated glossary with Monitor.

**Review by krga2022 — APPROVED** [review]

File: samples/wifi/monitor/README.rst [hunk 1]
```diff
@@ -0,0 +1,110 @@
+.. _wifi_monitor_sample:
+
+Wi-Fi: Monitor
```
Comment (bama-nordic) [inline samples/wifi/monitor/README.rst:3]: The way I see it, this sample should be a raw capture sample which can be configured in monitor mode and (later) promiscuous mode . Let's discuss this internally and modify accordingly.

File: samples/wifi/monitor/README.rst [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+.. table-from-sample-yaml::
+
+Overview
+********
+
+The sample demonstrates how to configure nrf70 Series device in Monitor mode.
+It analyzes the Wi-Fi incoming packets on a raw socket and prints the packet statistics at fixed internal.
+
+* :kconfig:option:`CONFIG_STATS_PRINT_TIMEOUT`: Wait duration for printing Wi-Fi packet statistics.
```
[the remark this answers is not in the document]
Comment (chiranjeevi2776) [reply samples/wifi/monitor/README.rst]: Done.

File: samples/wifi/monitor/README.rst (hunk 1 above)
Comment (chiranjeevi2776) [reply samples/wifi/monitor/README.rst:3]: Following internal discussions, maintain "Monitor" as a distinct example, and treat "promiscuous mode" as a separate illustration.

**Review by bama-nordic — APPROVED** [review]

**Review by VivekUppunda — APPROVED** [review]

File: samples/wifi/monitor/README.rst [hunk 3]
```diff
@@ -0,0 +1,125 @@
+.. _wifi_monitor_sample:
+
+Wi-Fi: Monitor
+##############
+
+.. contents::
+   :local:
+   :depth: 2
+
+The Monitor sample demonstrates how to set monitor mode, analyzing incoming Wi-Fi packets, and printing packet statistics.
```
Comment (richabp) [inline samples/wifi/monitor/README.rst:10]:
```suggestion
The Monitor sample demonstrates how to set monitor mode, analyze incoming Wi-Fi® packets, and print packet statistics.
```

File: samples/wifi/monitor/README.rst [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+************
+
+The sample supports the following development kits:
+
+.. table-from-sample-yaml::
+
+Overview
+********
+
+The sample demonstrates how to configure nrf70 Series device in Monitor mode.
```
Comment (richabp) [inline samples/wifi/monitor/README.rst:22]:
```suggestion
The sample demonstrates how to configure the nRF70 Series device in Monitor mode.
```

File: samples/wifi/monitor/README.rst [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+It analyzes the Wi-Fi incoming packets on a raw socket and prints the packet statistics at fixed internal.
```
Comment (richabp) [inline samples/wifi/monitor/README.rst:23]:
```suggestion
It analyzes the incoming Wi-Fi packets on a raw socket and prints the packet statistics at a fixed interval.
```

File: samples/wifi/monitor/README.rst [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+* :kconfig:option:`CONFIG_STATS_PRINT_TIMEOUT`: Wait duration for printing Wi-Fi packet statistics in seconds.
```
Comment (richabp) [inline samples/wifi/monitor/README.rst:25]:
```suggestion
To set the wait duration for printing Wi-Fi packet statistics in seconds, use the :kconfig:option:`CONFIG_STATS_PRINT_TIMEOUT` Kconfig option.
```

File: samples/wifi/monitor/README.rst [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+      [00:00:05.492,980] <inf> monitor: Data Frames:
+      [00:00:05.492,980] <inf> monitor:       Data Count: 5
+      [00:00:05.492,980] <inf> monitor:       QoS Data Count: 0
+      [00:00:05.492,980] <inf> monitor:       Null Count: 0
+      [00:00:05.493,011] <inf> monitor:       QoS Null Count: 0
+
+Dependencies
+************
+
+This sample uses the following library:
```
Comment (richabp) [inline samples/wifi/monitor/README.rst:123]:
```suggestion
This sample uses the following |NCS| library:
```

File: samples/wifi/monitor/README.rst [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+.. |sample path| replace:: :file:`samples/wifi/monitor`
+
+.. include:: /includes/build_and_run_ns.txt
+
+To build for the nRF7002 DK, use the ``nrf7002dk_nrf5340_cpuapp`` build target.
+The following are examples of the CLI commands:
+
+  .. code-block:: console
+
+     west build -b nrf7002dk_nrf5340_cpuapp
```
Comment (richabp) [inline samples/wifi/monitor/README.rst:51]:
```suggestion
The following is an example of the CLI command:

.. code-block:: console

   west build -b nrf7002dk_nrf5340_cpuapp
```

File: doc/nrf/glossary.rst [hunk 9]
```diff
@@ -442,6 +442,9 @@ Glossary
    Modulation Coding Scheme (MCS)
       An index defined by the Wi-Fi standards where each index specifies a combination of modulation scheme and forward error correction coding rate.
 
+   Monitor
+      A network interface mode that enables the capturing and analysis of all incoming IEEE 802.11 packets.
```
Comment (richabp) [inline doc/nrf/glossary.rst:446]:
```suggestion
      A network interface mode that enables the capture and analysis of all incoming IEEE 802.11 packets.
```

**Review by richabp — CHANGES_REQUESTED** [review]

**Review by D-Triveni — APPROVED** [review]

File: doc/nrf/glossary.rst (hunk 9 above)
Comment (greg-fer) [reply doc/nrf/glossary.rst:446]: Isn't this Wi-Fi specific? Would be good to mention this in the term definition, such as:
```suggestion
      A network interface mode for nRF70 Series devices that enables the capture and analysis of all incoming IEEE 802.11 packets.
```

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: samples/wifi/monitor/README.rst [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+.. table-from-sample-yaml::
+
+Overview
+********
+
+The sample demonstrates how to configure the nrf70 Series device in Monitor mode.
+It analyzes the incoming Wi-Fi packets on a raw socket and prints the packet statistics at a fixed interval.
+
+To set the wait duration for printing Wi-Fi packet statistics in seconds, use the
+:kconfig:option:`CONFIG_STATS_PRINT_TIMEOUT` Kconfig option.
```
Comment (richabp) [inline samples/wifi/monitor/README.rst:26]:
```suggestion
To set the wait duration for printing Wi-Fi packet statistics in seconds, use the :kconfig:option:`CONFIG_STATS_PRINT_TIMEOUT` Kconfig option.
```

**Review by richabp — CHANGES_REQUESTED** [review]

File: doc/nrf/glossary.rst (hunk 9 above)
Comment (richabp) [reply doc/nrf/glossary.rst]: @chiranjeevi2776 -. Please see this suggestion.

File: samples/wifi/monitor/README.rst (hunk 3 above)
Comment (richabp) [reply samples/wifi/monitor/README.rst]: **Wi-Fi®** seems to be missed.

File: samples/wifi/monitor/README.rst (hunk 4 above)
Comment (richabp) [reply samples/wifi/monitor/README.rst]: nRF- **RF** is capitalized.

**Comment by krga2022** [comment]
Only control frame counts are updating. Management frames and data frame counts is 0.

`[02:44:37.250,152] <inf> monitor: Management Frames:
[02:44:37.250,183] <inf> monitor:       Beacon Count: 0
[02:44:37.250,183] <inf> monitor:       Probe Request Count: 0
[02:44:37.250,183] <inf> monitor:       Probe Response Count: 0
[02:44:37.250,183] <inf> monitor: Control Frames:
[02:44:37.250,213] <inf> monitor:        Ack Count 17954
[02:44:37.250,213] <inf> monitor:        RTS Count 0
[02:44:37.250,213] <inf> monitor:        CTS Count 0
[02:44:37.250,213] <inf> monitor:        Block Ack Count 50
[02:44:37.250,213] <inf> monitor:        Block Ack Req Count 0
[02:44:37.250,244] <inf> monitor: Data Frames:
[02:44:37.250,244] <inf> monitor:       Data Count: 0
[02:44:37.250,244] <inf> monitor:       QoS Data Count: 0
[02:44:37.250,244] <inf> monitor:       Null Count: 0
[02:44:37.250,274] <inf> monitor:       QoS Null Count: 0`

**Comment by chiranjeevi2776** [comment]
> Only control frame counts are updating. Management frames and data frame counts is 0.
> 
> `[02:44:37.250,152] <inf> monitor: Management Frames: [02:44:37.250,183] <inf> monitor: Beacon Count: 0 [02:44:37.250,183] <inf> monitor: Probe Request Count: 0 [02:44:37.250,183] <inf> monitor: Probe Response Count: 0 [02:44:37.250,183] <inf> monitor: Control Frames: [02:44:37.250,213] <inf> monitor: Ack Count 17954 [02:44:37.250,213] <inf> monitor: RTS Count 0 [02:44:37.250,213] <inf> monitor: CTS Count 0 [02:44:37.250,213] <inf> monitor: Block Ack Count 50 [02:44:37.250,213] <inf> monitor: Block Ack Req Count 0 [02:44:37.250,244] <inf> monitor: Data Frames: [02:44:37.250,244] <inf> monitor: Data Count: 0 [02:44:37.250,244] <inf> monitor: QoS Data Count: 0 [02:44:37.250,244] <inf> monitor: Null Count: 0 [02:44:37.250,274] <inf> monitor: QoS Null Count: 0`

Check with latest ncs main..

File: samples/wifi/monitor/Kconfig [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+
+menu "Nordic Monitor sample"
+
+config MONITOR_MODE_CHANNEL
+	int "Configure the channel for Monitor mode sample"
+	default 1
+	help
+	  Specify the channel for receiving WiFi packets in Monitor mode.
+	  The valid channel range for 2G is 1 to 14
+	  The valid channel range for 5G starts from 36, 40, 44 up to 165.
```
Comment (richabp) [inline samples/wifi/monitor/Kconfig:17]:
```suggestion
	  Specify the channel for receiving Wi-Fi packets in Monitor mode.
	  The valid channel range for 2G is 1 to 14.
	  The valid channel range for 5G starts from 36, 40, 44, up to 165.
```

**Review by richabp — CHANGES_REQUESTED** [review]

**Review by richabp — APPROVED** [review]

**Pull request merged**